### PR TITLE
New user api version of SSOUSER,SSUUSER,MANAGEDUSER Create and UPDATE…

### PR DESCRIPTION
**Pull request by sreeragksgh**
…USER

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/kong-api/defaults/main.yml [hunk 1]
```diff
@@ -5665,3 +5665,74 @@ kong_apis:
...
+      uris: "{{ user_service_prefix }}/v1/create"
+      upstream_url: "{{ learning_service_url }}/v1/manageduser/create"
+      strip_uri: true
+      plugins:
+        - name: jwt
+        - name: cors
+        - "{{ statsd_pulgin }}"
+        - name: acl
+          config.whitelist:
+            - 'ManagedUserCreate'
```
Comment (Hari-stackroute) [inline ansible/roles/kong-api/defaults/main.yml:5696]: use whitelist value as "userAccess" which is mentioned for v4/create

**Review by Hari-stackroute — APPROVED** [review]

File: ansible/roles/kong-api/defaults/main.yml [hunk 2]
```diff
@@ -5665,3 +5665,124 @@ kong_apis:
...
+          config.whitelist:
+            - 'userUpdate'
+        - name: rate-limiting
+          config.policy: local
+          config.hour: "{{ medium_rate_limit_per_hour }}"
+          config.limit_by: credential
+        - name: request-size-limiting
+          config.allowed_payload_size: "{{ medium_request_size_limit }}"
+
+  - name: managedUserV1
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5686]: can we have meaningful api names? like managedUserV1Create or managedUserV1Update or managedUserV1Delete for all the below apis as well, the api names should indicate what the apis do @Hari-stackroute

Comment (Hari-stackroute) [reply ansible/roles/kong-api/defaults/main.yml]: @SMYALTAMASH : yes , sreerag will be doing the changes.

Comment (sreeragksgh) [reply ansible/roles/kong-api/defaults/main.yml]: Done

File: ansible/roles/kong-api/defaults/main.yml [hunk 3]
```diff
@@ -5665,3 +5665,124 @@ kong_apis:
...
+        config.whitelist:
+          - 'userTempAdmin'
+      - name: rate-limiting
+        config.policy: local
+        config.hour: "{{ medium_rate_limit_per_hour }}"
+        config.limit_by: credential
+      - name: request-size-limiting
+        config.allowed_payload_size: "{{ small_request_size_limit }}"
+
+  - name: searchOrgV2
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5776]: are we depricating orgSearch API?

File: ansible/roles/kong-api/defaults/main.yml [hunk 4]
```diff
@@ -5665,3 +5665,124 @@ kong_apis:
...
+          config.whitelist:
+            - 'userAccess'
+        - name: rate-limiting
+          config.policy: local
+          config.hour: "{{ medium_rate_limit_per_hour }}"
+          config.limit_by: credential
+        - name: request-size-limiting
+          config.allowed_payload_size: "{{ medium_request_size_limit }}"
+
+  - name: ssoUserV1Create
```
Comment (SMYALTAMASH) [inline ansible/roles/kong-api/defaults/main.yml:5704]: is this used by mobile devices?

Comment (sreeragksgh) [reply ansible/roles/kong-api/defaults/main.yml:5704]: only in portal

Comment (sreeragksgh) [reply ansible/roles/kong-api/defaults/main.yml:5704]: Hi @SMY-ALTAMASH  The documentation pr is ready hari will be merging that after he have a look in to it.

**Review by SMYALTAMASH — APPROVED** [review]

**Pull request merged**